File: Dot examples/ASP Book Examples/Book Examples/Chapter34/vb/App_Code/Controls/UserNameParameter.vb
Imports System
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls


Namespace Superexpert.Controls

    ''' <summary>
    ''' Summary description for UserNameParameter
    ''' </summary>
    Public Class UserNameParameter
     Inherits Parameter
        Protected Overrides Function Evaluate(ByVal context As HttpContext, ByVal control As Control) As Object
            Return context.User.Identity.Name
        End Function

    End Class

End Namespace

File: Dot examples/ASP Book Examples/Book Examples/Chapter34/cs/App_Code/Controls/UserNameParameter.cs
using System;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;


namespace Superexpert.Controls
{

    /// <summary>
    /// Summary description for UserNameParameter
    /// </summary>
    public class UserNameParameter : Parameter
    {
        protected override object Evaluate(HttpContext context, Control control)
        {
            return context.User.Identity.Name;
        }

    }

}